### PR TITLE
Add json+ld image support for individual entries

### PR DESCRIPTION
**Pull request by christianc1**

### CODE REVIEW

File: classes/class-wpcom-liveblog-entry.php [hunk 1]
```diff
@@ -535,6 +535,38 @@ public static function get_entry_title( $entry ) {
...
+	 * Get Entry first image if present.
+	 *
+	 * @param  object $entry Entry.
+	 * @return string        Image src.
+	 */
+	public static function get_entry_first_image( $entry ) {
+		$entry_id = ( $entry instanceof WPCOM_Liveblog_Entry ) ? $entry->get_id() : $entry->id;
+		$entry_content = ( $entry instanceof WPCOM_Liveblog_Entry ) ? $entry->get_content() : $entry->content;
+		$key    = 'liveblog_entry_' . $entry_id . 'first_image';
+		$cached = wp_cache_get( $key, 'liveblog' );
```
Comment (paulschreiber) [inline classes/class-wpcom-liveblog-entry.php:548]: We'll need to invalidate this cache if the entry is updated or deleted.

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: classes/class-wpcom-liveblog-entry.php [hunk 2]
```diff
@@ -535,6 +535,38 @@ public static function get_entry_title( $entry ) {
...
 	}
 
+	/**
+	 * Get Entry first image if present.
+	 *
+	 * @param  object $entry Entry.
+	 * @return string        Image src.
+	 */
+	public static function get_entry_first_image( $entry ) {
+		$entry_id = ( $entry instanceof WPCOM_Liveblog_Entry ) ? $entry->get_id() : $entry->id;
```
Comment (paulschreiber) [inline classes/class-wpcom-liveblog-entry.php:545]: If $entry isn't a `WPCOM_Liveblog_Entry`, what is it instead?

Comment (christianc1) [reply classes/class-wpcom-liveblog-entry.php:545]: @paulschreiber ,

`WPCOM_Liveblog::get_entries_paged` returns an array of `stdClass` objects with data as properties.

File: classes/class-wpcom-liveblog-entry.php (hunk 1 above)
Comment (christianc1) [reply classes/class-wpcom-liveblog-entry.php:548]: Update/delete entries are stored as new comments, and have their own `id`s, so there isn't a need to invalidate as the data would never be stale.  Granted, once an entry is updated, we could delete the old cache entry and prime the cache with the updated entry, but it's not necessary.  Provided the above, do you have a recommendation on what you'd like pursued?

Comment (paulschreiber) [reply classes/class-wpcom-liveblog-entry.php:548]: Wow, that's super-weird. Given that, we don't need to do anything.

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**